src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +46,35 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (properTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean properTeamSize(EntityRef teleporter, EntityRef player) {
+        int maxTeamSizeDifference = 3;
+        int oppositeTeamCount = 0;
+        int teleporterTeamCount = 0;
```

Comment:
IMO we should track this more globally instead of counting again every time a player tries to join a team.

Reply:
Done